### ISSUE TITLE
[Refactor] Make the paint functions as part of the PaintSession structure

### ISSUE THREAD

**Issue by frutiemax** [OpenRCT2/OpenRCT2#18761]
Currently, those paint functions are in the global namespace:

- PaintAddImageAsParent
- PaintAddImageAsChild
- PaintAddImageAsChildRotated
- PaintUtilPushTunnelRotated
- PaintAttachToPreviousAttach
- PaintAttachToPreviousPS
- PaintFloatingMoneyEffect
- PaintDrawStructs

These functions should be part of the PaintSession structure as this will enable the use of Google Mock and validate parameters passed in those functions (offsets, bounding boxes, etc.) using unit tests.

**Comment by frutiemax** [comment]
After thought, there are more than 30k lines of code that are impacted by this, I'm not sure if this is a good idea after all...

**Comment by Gymnasiast** [comment]
I wouldn’t embark on something this big without our input anyway.

**Comment by ZehMatt** [comment]
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c9-minimize-exposure-of-members

**Comment by Gymnasiast** [comment]
@ZehMatt Do you think this is a good idea?

**Comment by ZehMatt** [comment]
Not necessarily, PaintSession should be renamed to PaintGraph/SceneGraph, it should only contain the resulting data from the PaintGraphBuilder/SceneGraphBuilder. The game currently uses one graph per column, the columns are the screen width divided by 32, it does take neighboring columns into account when building the graph.

So the usage should become something like this: 
```cpp
auto builder = PaintGraphBuilder(viewportData, options);

// PaintSessionGenerate
builder.add(image, offset, bounds);
auto* root = builder.add(image2, offset2, bounds2);
builder.addChild(root, ...);
..

// PaintSessionArrange
auto graph = builder.finish(); // Sort the graph and result it.

// ViewportPaintColumn
draw(graph);
```

Doing something like will also enable to have a much better capabilities to add tests for this, one of the issues will probably be memory management, we have currently a pretty hack solution to share a single pool among multiple threads but with the separation of state we no longer need that much memory as the resulting graph is what matters and the state of the builder is temporary which is currently all sitting in PaintSession.
